README.md

```diff
@@ -16,55 +16,57 @@ First, you'll need a `vercel.json` file in your project:
 {
   "functions": {
     "api/**/*.rs": {
-      "runtime": "vercel-rust@3.1.2"
+      "runtime": "vercel-rust@4.0.0"
     }
   }
 }
 ```
 
-A Vercel Function will be created for every file that matches `api/**/*.rs`. Next, you can create a new Function `api/user.rs`:
+A Vercel Function will be created for every file that matches `api/**/*.rs`.
+
+Example:
 
 ```rust
-use http::{StatusCode};
-use vercel_lambda::{lambda, error::VercelError, IntoResponse, Request, Response};
-use std::error::Error;
-
-fn handler(_: Request) -> Result<impl IntoResponse, VercelError> {
-	let response = Response::builder()
-		.status(StatusCode::OK)
-		.header("Content-Type", "text/plain")
-		.body("Hello World")
-		.expect("Internal Server Error");
-
-		Ok(response)
+use serde_json::json;
+use vercel_runtime::{
+    lambda_http::{http::StatusCode, Error as LambdaError, Response},
+    run, IntoResponse, ProxyError, ProxyRequest,
+};
+
+#[tokio::main]
+async fn main() -> Result<(), LambdaError> {
+    run(handler).await?;
+    Ok(())
 }
 
-// Start the runtime with the handler
-fn main() -> Result<(), Box<dyn Error>> {
-	Ok(lambda!(handler))
+pub async fn handler(_req: ProxyRequest) -> Result<impl IntoResponse, ProxyError> {
+    let response = Response::builder()
+        .status(StatusCode::OK)
+        .header("Content-Type", "application/json")
+        .body(
+            json!({
+              "message": "你好，世界"
+            })
+            .to_string(),
+        )?;
+
+    Ok(response)
 }
 ```
 
-Finally, we need an `api/Cargo.toml` file:
+Finally we need a `Cargo.toml` file at the root of your repository.
 
 ```toml
-[package]
-name = "index"
-version = "1.0.0"
-authors = ["Your Name <your@site.com>"]
-edition = "2018"
-
-[dependencies]
-http = "0.1"
-vercel_lambda = "*"
-
-[lib]
-name = "util"
-path = "_util.rs"
+# You can specify a library for shared logic here (optional)
+# [lib]
+# path = "src-rs/lib.rs"
+
+# Each handler has to be specified as [[bin]]
+[[bin]]
+name = "handler"
+path = "api/handler.rs"
```

Comment:
I think we need to find a solution to doing this via `build.rs` or similar. Or maybe some regular `src/main.rs` that you drop a macro in, and it pulls all the handlers from `/api/...`

Reply:
I thought of that as well, issue being that then rust analyzer doesn't pick up the files during development.